### PR TITLE
[@types/node] Add crypto to globalThis

### PR DESCRIPTION
**Pull request by g12i**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/webcrypto.html#class-crypto
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by jakebailey** [comment]
This is missing the change in the ts4.8 dir.

File: types/node/crypto.d.ts [hunk 1]
```diff
@@ -4457,6 +4457,12 @@ declare module "crypto" {
             ): Promise<ArrayBuffer>;
         }
     }
+
+    global {
+        namespace NodeJS {
+            const crypto: webcrypto.Crypto;
```
Comment (Semigradsky) [inline types/node/crypto.d.ts:4463]: Why did you add `crypto` to the `NodeJS` namespace?

Comment (g12i) [reply types/node/crypto.d.ts:4463]: To be honest, I simply adhered to what appeared to be a convention, placing global Node-related elements within this namespace

Comment (Semigradsky) [reply types/node/crypto.d.ts:4463]: Only interfaces or types can be in `NodeJS` namespace since there is no `NodeJS` variable in global context:
```
> console.log(global.NodeJS)
undefined
```

File: types/node/test/globals.ts [hunk 2]
```diff
@@ -88,3 +88,17 @@ declare var RANDOM_GLOBAL_VARIABLE: true;
     const f: File = {} as any;
     f.name; // $ExpectType string
 }
+
+{
+    crypto.randomUUID(); // $ExpectType `${string}-${string}-${string}-${string}-${string}` || string
```
Comment (Semigradsky) [inline types/node/test/globals.ts:93]: Here you used imported `crypto`, not from global context. Use `globalThis.crypto`

Comment (Semigradsky) [reply types/node/test/globals.ts:93]: Oh, sorry, I confused with `test/crypto.ts`...

**Review by Semigradsky — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
@g12i thanks!
Keep an eye over here for couple of days, once released to NPM

**Comment by g12i** [comment]
Ready to merge

**Comment by Semigradsky** [comment]
Ready to merge

**Comment by lenovouser** [comment]
I am getting the following error now when compiling:

```
node_modules/@types/node/crypto.d.ts:4478:13 - error TS2502: 'crypto' is referenced directly or indirectly in its own type annotation.

4478         var crypto: typeof globalThis extends {
                 ~~~~~~


Found 1 error in node_modules/@types/node/crypto.d.ts:4478
```

**Comment by Semigradsky** [comment]
@lenovouser What the version of TypeScript you have? Also please check that you have only one `@types/node` in `node_modules`

**Comment by lenovouser** [comment]
TypeScript 5.3.3 - there should be only one @types/node, but dependencies might use different versions? I am using pnpm in case that helps.

```
➜ pnpm why @types/node
Legend: production dependency, optional only, dev only

devDependencies:
@types/iarna__toml 2.0.5
└── @types/node 20.11.8
@types/node 20.11.8
tsup 8.0.1
└─┬ postcss-load-config 4.0.2
  └─┬ ts-node 10.9.2 peer
    └── @types/node 20.11.8 peer
zx 7.2.3
├─┬ @types/fs-extra 11.0.4
│ ├─┬ @types/jsonfile 6.1.4
│ │ └── @types/node 20.11.8
│ └── @types/node 20.11.8
└── @types/node 18.19.10
```

**Comment by mkorman9** [comment]
I've created an empty project with @types/node 20.11.8, and it seems it's the only version in node_modules
```
> npm list --depth=1000 | grep '@types/node'
├─┬ @types/node@20.11.8
```

But I still got the error
```
node_modules/@types/node/crypto.d.ts:4478:13 - error TS2502: 'crypto' is referenced directly or indirectly in its own type annotation.

4478         var crypto: typeof globalThis extends {
```

**Comment by jakebailey** [comment]
Reading this PR back, I don't think this PR is correct. The declarations between 4.8 and the non 4.8 folder should be identical; the only reason that the two folders exist at the moment is that one tests DOM and one does not (which is not ideal as we know, but the only method we have right now).

What this PR did was make it pass when DOM was present... in only one of them. But that's not how it works on user machines.

**Comment by jakebailey** [comment]
Sent https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68371
